config/v1/types_infrastructure.go

```diff
@@ -144,6 +148,24 @@ type GCPPlatformStatus struct {
 	Region string `json:"region"`
 }
 
+// BareMetalPlatformStatus holds the current status of the BareMetal infrastructure provider.
+type BareMetalPlatformStatus struct {
+	// apiInternalIP is an IP address to contact the Kubernetes API server that can be used
+	// by components inside the cluster, like kubelets using the infrastructure rather
+	// than Kubernetes networking. It is the IP that the Infrastructure.status.apiServerInternalURI
+	// points to. It is the IP for a self-hosted load balancer in front of the API servers.
+	APIInternalIP string `json:"apiInternalIP,omitempty"`
+
+	// ingressIP is an external IP which routes to the default ingress controller.
+	// The IP is a suitable target of a wildcard DNS record used to resolve default route host names.
+	IngressIP string `json:"ingressIP,omitempty"`
+
+	// nodeDNSIP is the IP address for the internal DNS used by the
```

Comment:
Yes. It is for the node hostnames. Since baremetal does not have a DNSaaS, we want to minimize the amount of external datacenter DNS entries that would be necessary for a deployer to have to create.

Reply:
will this IP return the address for `api-int.cluster_domain` DNS name ?? or there's no change this will return the addressed for non-node-hostnames.